src/pages/billing/Project.js

```diff
@@ -452,6 +421,41 @@ const ProjectDetail = ({ authorizeAndLoadAccounts, billingAccounts, billingProje
   useOnMount(() => { reloadBillingProjectUsers() })
 
   useEffect(() => { StateHistory.update({ projectUsers }) }, [projectUsers])
+  // Update cost data only if report date range changes, or if spend report tab was selected.
+  useEffect(() => {
+    const maybeLoadTotalCost = _.flow(
+      reportErrorAndRethrow('Unable to retrieve spend report data'),
+      Utils.withBusyState(setUpdating)
+    )(async () => {
+      if (!updatingTotalCost && totalCost === null && tab === spendReportKey) {
+        setUpdatingTotalCost(true)
+        const endDate = new Date().toISOString().slice(0, 10)
+        const startDate = new Date((Date.now() - spendReportLengthInDays * 24 * 60 * 60 * 1000)).toISOString().slice(0, 10)
```

Comment:
This is going to be flaky with DST (however long that stays with us ;) ). Is there a more idiomatic way in JS to add/subtract days?

Reply:
I can look into it.

Reply:
We have https://date-fns.org/as a dependency already

Reply:
Thanks, I will check it out, but would like to go ahead and merge this PR as-is since it doesn't introduce this problem.

@zarsky-broad or @dmohs if I can get a thumbs on this PR it will help me to move the review forward on a dependent PR (it will make it easier to see what has actually changed in the next PR if this refactoring is merged first).

Reply:
It looks like @dmohs is on vacation, so I'm removing him as a reviewer. @zarsky-broad, it's all up to you now. :)